autoload/lsp/ui/vim.vim

```diff
@@ -505,14 +538,46 @@ function! s:handle_code_action(server, last_req_id, type, data) abort
     let l:choice = inputlist(l:choices)
 
     if l:choice > 0 && l:choice <= l:index
-        call lsp#log('s:handle_code_action', l:codeActions[l:choice - 1]['arguments'][0])
-        call s:apply_workspace_edits(l:codeActions[l:choice - 1]['arguments'][0])
+        call s:execute_command_or_code_action(a:server, l:codeActions[l:choice - 1])
     endif
 endfunction
 
+" @params
+"   server - string
+"   comand_or_code_action - Command | CodeAction
+function! s:execute_command_or_code_action(server, command_or_code_action) abort
+    if has_key(a:command_or_code_action, 'command') && type(a:command_or_code_action['command']) == type('')
+        let l:command = a:command_or_code_action
+        call s:execute_command(a:server, l:command)
+    else
+        let l:code_action = a:command_or_code_action
+        if has_key(l:code_action, 'edit')
+            call lsp#ui#vim#apply_workspace_edits(a:command_or_code_action['edit'])
+        endif
+        if has_key(l:code_action, 'command')
+            call s:execute_command(a:server, l:code_action['command'])
+        endif
+    endif
+endfunction
+
+" Sends workspace/executeCommand with given command.
+" @params
+"   server - string
+"   command - https://microsoft.github.io/language-server-protocol/specification#command
+function! s:execute_command(server, command) abort
+    let l:params = {'command': a:command['command']}
+    if has_key(a:command, 'arguments')
+        let l:params['arguments'] = a:command['arguments']
+    endif
+    call lsp#send_request(a:server, {
+        \ 'method': 'workspace/executeCommand',
+        \ 'params': l:params,
+        \ })
+endfunction
+
 " @params
 "   workspace_edits - https://microsoft.github.io/language-server-protocol/specification#workspaceedit
-function! s:apply_workspace_edits(workspace_edits) abort
+function! lsp#ui#vim#apply_workspace_edits(workspace_edits) abort
```

Comment:
Can we move this function to `lsp#ui#vim#workspace_edits#apply_workspace_edits` similar to how we have `lsp#utils#text_edit#apply_text_edits`

Reply:
I moved it to `lsp#utils#workspace_edits#apply_workspace_edit` to match `text_edit`. Please let me know if you would prefer `lsp#ui#vim` over `lsp#utils`.

Reply:
`lsp#utils` is better.